fcrepo-kernel-modeshape/src/test/java/org/fcrepo/kernel/modeshape/UrlBinaryTest.java

```diff
@@ -0,0 +1,285 @@
+
+    private FedoraBinary testObj;
+
+    private String mimeType;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(options().dynamicPort());
+
+    private String fileUrl;
+
+    @Mock
+    private Session mockSession;
+
+    @Mock
+    private Property mimeTypeProperty;
+
+    @Mock
+    private Property proxyURLProperty;
+
+    @Mock
+    private Property redirectURLProperty;
+
+    @Mock
+    private Property mockProperty;
+
+    @Mock
+    private Value mockValue;
+
+    @Mock
+    private Value mockURIValue;
+
+    @Mock
+    private Node mockDescNode, mockContent, mockParentNode;
+
+    @Mock
+    private NodeType mockDescNodeType;
+
+    @Mock
+    private InputStream mockStream;
+
+    @Before
+    public void setUp() throws Exception {
+        final NodeType[] nodeTypes = new NodeType[] { mockDescNodeType };
+
+        stubFor(get(urlEqualTo("/file.txt"))
+                .willReturn(aResponse()
+                        .withHeader("Content-Type", "text/plain")
+                        .withBody(EXPECTED_CONTENT)));
+        fileUrl = "http://localhost:" + wireMockRule.port() + "/file.txt";
+
+        mimeType = "text/plain";
+
+        when(mockDescNodeType.getName()).thenReturn(FEDORA_NON_RDF_SOURCE_DESCRIPTION);
+        when(mockDescNode.getMixinNodeTypes()).thenReturn(nodeTypes);
+        when(mockDescNode.getParent()).thenReturn(mockContent);
+        when(mockContent.getSession()).thenReturn(mockSession);
+        when(mockContent.isNodeType(FEDORA_BINARY)).thenReturn(true);
+        when(mockContent.getParent()).thenReturn(mockParentNode);
+        when(mockContent.getNode(FEDORA_DESCRIPTION)).thenReturn(mockDescNode);
+        when(mockDescNode.setProperty(anyString(), any(Binary.class))).thenReturn(mockProperty);
+
+        final NodeType mockNodeType = mock(NodeType.class);
+        when(mockNodeType.getName()).thenReturn("nt:versionedFile");
+        when(mockContent.getPrimaryNodeType()).thenReturn(mockNodeType);
+
+        testObj = new UrlBinary(mockContent);
+    }
+
+    @Test
+    public void testSetContent() throws Exception {
+        mockProxyProperty();
+        testObj.setContent(mockStream, mimeType, null, null, null);
+
+        verify(mockDescNode).setProperty(HAS_MIME_TYPE, mimeType);
+    }
+
+    @Test
+    public void testSetContentWithFilename() throws Exception {
+        mockProxyProperty();
+        final String fileName = "content.txt";
+        testObj.setContent(mockStream, mimeType, null, fileName, null);
+
+        verify(mockDescNode).setProperty(HAS_MIME_TYPE, mimeType);
+        verify(mockDescNode).setProperty(FILENAME, fileName);
+    }
+
+    @Test
+    public void testSetContentWithChecksum() throws Exception {
+        final String checksum = checksumString(EXPECTED_CONTENT);
+        mockProxyProperty();
+
+        testObj.setContent(mockStream, mimeType, singleton(
+                new URI(checksum)), null, null);
```

Comment:
There doesn't seem to be any verification in this test.

Reply:
Fixed this in on of my later commits  - though it's not showing up here.